### DISCUSSION TITLE
how can I speed up archive snippet load

### DISCUSSION THREAD

**Discussion by xenoterracide** [z-shell/zi#295] [Q&A] [answered]
So, I'm installing maven with a snippet

```zsh
zi ice wait has'java' as'program' pick'bin/mvn' extract'!apache-maven-3.9.5-bin.
tar.gz'
zi snippet https://dlcdn.apache.org/maven/maven-3/3.9.5/binaries/apache-maven-3.
9.5-bin.tar.gz
```

on the initial terminal (I don't mean first ever, I mean every time I open a new terminal/tab) load I can always see the "loaded" snippet/url show up briefly. any way I can put an end to that? I'm not entirely certain what it's doing there.

**Comment by ss-o** [comment, marked as answer]
You are missing the ice modifier `lucid` :dart: 

- https://wiki.zshell.dev/docs/getting_started/overview#turbo-mode--lucid
- https://wiki.zshell.dev/docs/guides/syntax/for#lucid
- https://wiki.zshell.dev/docs/guides/syntax/ice-modifiers#lucid

**Reply by xenoterracide** [reply]
always been confusing to me when to use lucid and not. I sometimes think about switching from `zi` because I don't find it super intuitive. Then I think about the effort to switch.

**Reply by ss-o** [reply]
I fully agree, but it is because of bad decisions from maintainers in the past. I striped a huge amount of code that was not required or deprecated and fixed many API-related issues. I struggle to find enough time to implement changes and new ideas as it requires a huge amount of time, however, it is an ongoing process, much work has been done and hopefully soon will start releasing it for testing. :rocket:
